reports/aksversions/README.md

```diff
@@ -8,50 +8,62 @@ The scripts goal is to find every AKS cluster within a set of subscriptions, fin
-    - Will then add all new documents supplied to the script from the `aks-versions.sh` script
+    - Will then add all new documents supplied to the script from the `main.py` script
 
 ## Dockerfile
 
-The Dockerfile will build an image that contains the scripts and will run the `aks-versions.sh` script when launched.
+The Dockerfile will build an image that contains the scripts and will run the `main.py` script when launched.
 This container image will be run as a cronjob so that it runs on a schedule and when complete the pod will stop and eventually be removed.
 
 The Dockerfile does not have any effect on the report process and is simply a way to make this deployable to AKS.
 
 ## Local dev
 
-It is possible to run this script locally if you have:access to the Azure as you will need to:
+As this report utilises Python you will need to have it installed, development was carried out using `Python 3.11.7` so a version greater than `3.11.x` is recommened.
 
-- Log into Azure via the `azure cli` so the script can use your credentials/access to read the subscriptions and AKS cluster information.
+The script utilises `DefaultAzureCredential` to access Azure.
+<br>Locally you simply need to log into Azure with the `azure-cli` using `az login` and the script will use your local permissions to access Azure.
 
-and also set the following environment variables:
+The `save-to-cosmos.py` script also requires access to the Cosmos DB account and these should be supplied as environment variables:
 
 - COSMOS_DB_URI
 - COSMOS_KEY
 
-Both of these are values that can be found via the Azure Portal on the version reporter Cosmos DB instance.
+These values can be found via the Azure Portal on the version reporter Cosmos DB instance.
 
-When setup you can simply run the script locally by using `./aks-versions.sh` from the `reports/aksVersions` directory.
+When setup you can run the script locally by using `python main.py` from the `reports/aksversions` directory.
+
+### Azure Python SDK
+
+The following links may be useful if changes to the script are required:
+
+[AKS Managed Cluster API (includes examples for Python)](https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters?view=rest-aks-2024-02-01)
+[AKS Managed Cluster class](https://learn.microsoft.com/en-us/python/api/azure-mgmt-containerservice/azure.mgmt.containerservice.v2022_01_01.models.managedcluster?view=azure-python)
 
 ### Making it safer
 
-Its also possible to completely ignore the Cosmos DB update when developing locally by commenting out the following line from the `aks-versions.sh` script:
+Its also possible to completely ignore the Cosmos DB update when developing locally by commenting out the following line from the `main.py` script:
 
-```bash
-store_document "$documents"
+```python
+try:
+    result = subprocess.run(["python", script] + args, check=True)
+except subprocess.CalledProcessError as e:
+    print(f"Script {script} failed with error: {e}")
 ```
```

Comment:
Is this still correct?
Looks like this https://github.com/hmcts/version-reporter-services/blob/439c389869ba13250bd4868da0cb2e8bbbcf5499/reports/aksversions/main.py#L139-L147 would be the bit to comment out now?

Reply:
Updated and added env var to allow disabling the cosmos setup easier